tools/debugging/extract_sp_exceptions.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+SCENARIO_REMOTE_URL="http://68.183.70.168:8000/scenario-player/scenarios/"
+CURL_COMMAND="curl --no-progress-meter"
+WGET_DIR="wget --no-parent --no-clobber --continue -nH -r"
+
+COLUMNS=$(/usr/bin/tput cols)
+
+### STYLES: IGNORE
+RESET="\e[0m"
+BOLD="\e[1m"
+###
+
+
+function display_usage {
+    echo "This script does the following:"
+    echo "1. Figure out the run number of each of the existing scenarios"
+    echo "2. Download the logs for each of the nodes involved in the latest scenario run"
+    echo "3. Looks through the logs and searches for errors / exceptions and reports them"
+    echo -e "\n\nUSAGE: extrach_sp_exceptions.sh [DESTINATION_DIR]\n"
+    echo -e "Where\n"
+    echo -e "\tDESTINATION_URL is optional and is used to specify the path prefix for the downloaded files. Otherwise the current directory is used\n"
+}
+
+function print_bold {
+    echo -e "${BOLD}$1${RESET}"
+}
+
+# if less than two arguments supplied, display usage 
+if [[  $1 == "--help" ]]; then 
+    display_usage
+    exit 1
+fi
+
+if [[ $1 != "" ]]; then
+    DESTINATION_DIR=$1
+else
+    DESTINATION_DIR=$(pwd)
+fi
+
+print_bold "Downloading PFS logs"
+
+ssh root@services-dev.raiden.network 'cd raiden-services/deployment/; docker-compose logs pfs-goerli | gzip' > ${DESTINATION_DIR}/pfs.log.gz
+
```

Comment:
You can check here for ssh command failure and stop the script. Something like

```sh
if [[ $? -ne 0 ]]; then
    echo "Error: ssh to server failed"
    exit 1
fi
```

Same for other commands. But may be an overkill.

Reply:
Added